### PR TITLE
Refactor BootstrapCheckTests to use expectThrows

### PR DESCRIPTION
**Pull request by jasontedor**
This commit modifies all uses of the pattern try/catch/assert in
BootstrapCheckTests to use expectThrows/assert.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java [hunk 1]
```diff
@@ -155,15 +156,15 @@ long getMaxHeapSize() {
             }
         };
 
-        try {
-            BootstrapCheck.check(true, Collections.singletonList(check), "testHeapSizeCheck");
-            fail("should have failed to initial heap size not equal to max heap size");
-        } catch (final RuntimeException e) {
+        {
+            final RuntimeException e =
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java]: Why a block per? You can often get away with something like
```java
Exception e;
e = expectThrows(RuntimeException.class, () -> BootstrapCheck.check(true, Collections.singletonList(check), "testHeapSizeCheck"));
assertEquals(e.getMessage(), "words");
e = expectThrows(RuntimeException.class, () -> BootstrapCheck.check(true, Collections.singletonList(check), "testHeapSizeCheck"));
assertEquals(e.getMessage(), "words");
e = expectThrows(RuntimeException.class, () -> BootstrapCheck.check(true, Collections.singletonList(check), "testHeapSizeCheck"));
assertEquals(e.getMessage(), "words");
```

Comment (nik9000) [inline core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java]: Like, probably with new lines between each one or something.

File: core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java [hunk 2]
```diff
@@ -155,15 +156,15 @@ long getMaxHeapSize() {
...
+                    expectThrows(
+                            RuntimeException.class,
+                            () -> BootstrapCheck.check(true, Collections.singletonList(check), "testHeapSizeCheck"));
             assertThat(
                     e.getMessage(),
                     containsString("initial heap size [" + initialHeapSize.get() + "] " +
-                            "not equal to maximum heap size [" + maxHeapSize.get() + "]")
-            );
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java]: +1 on that reformatting.

File: core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java (hunk 1 above)
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java]: They are not needed here so I pushed 07a8fcc2cb9cb68bf7bfb3d6aa6cd13e8b87b252.

**Comment by jasontedor** [comment]
@nik9000 Thanks for reviewing, I pushed a commit responding to your feedback.

**Comment by nik9000** [comment]
Lgtm

**Pull request closed without merging**